engine-tests/src/test/java/org/terasology/engine/persistence/internal/StorageManagerTest.java

```diff
@@ -147,7 +145,7 @@ public void setup(@TempDir Path tempHome) throws Exception {
 
         context.put(ChunkProvider.class, mock(ChunkProvider.class));
         WorldProvider worldProvider = mock(WorldProvider.class);
-        when(worldProvider.getWorldInfo()).thenReturn(new WorldInfo());
+        // when(worldProvider.getWorldInfo()).thenReturn(new WorldInfo());
```

Comment:
is this still required or can it be removed entirely?

Reply:
This is the thing that spooks me about mocks.

Mockito told me that not all the mocks were used for all the tests. For instance, some of the tests don't use the NetworkManager mocks. A couple of these lines weren't used by _any_ of the tests—those are the ones I've commented out.

But I have a hunch that they were needed at some point or they wouldn't have been written. What if the next test needs them? Or what if something somewhere in the depths of one of these systems shifts which methods it calls when, and now this gets called again?

So I left it a little messy. With the intent that we'll come back to this after #5010 is merged and take another look at what kind of HeadlessEnvironment these tests need.

Reply:
in that case, please add a TODO there so that it's clear, why this is commented-out but not removed a.k.a what's unclear about it and what's necessary before it can be clarified and either re-introduced or removed.